src/ddspdrum/torchmodule.py

```diff
@@ -485,3 +485,189 @@ def _forward(self, control_in: T, audio_in: T) -> T:
 
         audio_in = fix_length(audio_in, len(control_in))
         return control_in * audio_in
+
+
+class TorchSVF(TorchSynthModule):
+    """
+    A State Variable Filter that can do low-pass, high-pass, band-pass, and
+    band-reject filtering. Allows modulation of the cutoff frequency and an
+    adjustable resonance parameter. Can self-oscillate to make a sinusoid
+    oscillator.
+
+    Parameters
+    ----------
+
+    mode (str)              :   filter type, one of LPF, HPF, BPF, or BSF
+    cutoff (float)          :   cutoff frequency in Hz must be between 5 and
+                                half the sample rate. Defaults to 1000Hz.
+    resonance (float)       :   filter resonance, or "Quality Factor". Higher
+                                values cause the filter to resonate more. Must
+                                be greater than 0.5. Defaults to 0.707.
+    mod_depth (float)       :   Amount of modulation to apply to the cutoff from
+                                the control input during processing. Can be negative
+                                or positive in Hertz. Defaults to zero.
+    """
+
+    def __init__(
+        self,
+        mode: str,
+        cutoff: float = 1000.0,
+        resonance: float = 0.707,
+        mod_depth: float = 0.0,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+
+        # Set the filter type
+        self.mode = mode.lower()
+        assert mode in ['lpf', 'hpf', 'bpf', 'bsf']
+
+        nyquist = self.sample_rate / 2.0
+        self.add_parameters(
+            [
+                TorchParameter(
+                    value=cutoff,
+                    parameter_range=ParameterRange(5.0, nyquist, "log"),
+                    parameter_name="cutoff"
+                ),
+                TorchParameter(
+                    value=resonance,
+                    parameter_range=ParameterRange(0.01, 1000.0, "log"),
+                    parameter_name="resonance"
```

Comment:
Should this lower limit be 0.5 ?

Reply:
So it really just needs to be greater than 0 -- I set it to 0.5 kind of arbitrarily in the numpy version -- 0.01 gives it quite a bit more room for less intense resonance and wider bandpasses / bandstops